pkg/storage/unified/search/document.go

```diff
@@ -43,6 +43,14 @@ func (s *StandardDocumentBuilders) GetDocumentBuilders() ([]resource.DocumentBui
 				dsinfo = append(dsinfo, info)
 			}
 		}
+
+		// Fetch dashboard sprinkles for the namespace
+		// This could take a while if namespace has a lot of dashboards
+		stats, err := s.sprinkles.GetStats(ctx, namespace)
```

Comment:
This will save the stats on the builder which gets stored in the builderCache. Not sure if we should be worried about memory usage here. Might need to be more aggressive with the cache max size?

Reply:
ya -- we may also need to structure this so we can indicate when the initial load is done and safely clear expire the cache.  Lets start with with just getting it working and an LRU cache